### PR TITLE
`Exam mode`: Improve usability of attendance check

### PR DESCRIPTION
**Pull request by pellumbbaboci**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Exam attendance check feature is not easy to use and the UI is difficult to understand.

### Description

- Instead of empty list in Verify Attendance, show a green message that no issues appeared
- If there is an issue, also show how many issues appear
- Merge columns Planned Room  Planned Seat  Actual Room  Actual Seat  into two columns Room and Seat
- New column with a status of the attendance check (Not attended, Attendance Check failed , Attendance Check passed)
- only show status column after exam has finished

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Course with exam

1. Log in to Artemis
2. Navigate to Course Administration
3. Register students to exam 
4. Upload the pdf that contains student images, check if images are saved correctly.
5. Generate Student Exams (check if you can fetch all student exams and view specific student exam) 
6. Login as student 
7. Participate in exam and submit the exam
8. Log as instructor back and verify attendance (You should see students that started the exam but did not sign in a table. )
9. If all students where checked, you should see a message that all students were checked successfully. 
10. Also in the students table you should see a new column called status after the exam has ended (statuses: Not attended, Attendance Check failed , Attendance Check passed)

Example files: 
[studentsWithImages.pdf](https://github.com/ls1intum/Artemis/files/10910361/studentsWithImages.pdf)
[studentsArtemis (1).csv](https://github.com/ls1intum/Artemis/files/10910367/studentsArtemis.1.csv)

Note:In Order for students to sign, you need to have IPad and the exam-attendance-checker app downloaded. But if you want you can change/update exam-user information by making a request to the following endpoint (use postman): 

``` 
/**
     * POST /courses/:courseId/exams/:examId/exam-users : Update the exam user with the given DTO info
     *
     * @param examUserDTO   the dto containing exam user info
     * @param signatureFile the signature of the student
     * @param courseId      the id of the course
     * @param examId        the id of the exam
     * @return saved examUser ResponseEntity with status 200 (OK) or with status 404 (Not Found)
     */
    @PostMapping("courses/{courseId}/exams/{examId}/exam-users")
    @PreAuthorize("hasRole('INSTRUCTOR')")
    public ResponseEntity<ExamUser> updateExamUser(@RequestPart ExamUserDTO examUserDTO, @RequestPart(value = "file", required = false) MultipartFile signatureFile,
            @PathVariable Long courseId, @PathVariable Long examId) {

....................
}
 ```
Here you can see an postman request example: 

exampleOfPostman
examUserDTO object is as follow:

```
{
 "login": "student",
"didCheckImage":true,
"didCheckLogin":true,
"didCheckName":true,
"didCheckRegistrationNumber":true
}
```
Note: login field is important as its used as identifier for students

#### Performance Review
- [ ] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Students page before:

Screenshot 2023-04-28 at 09 53 50

Students page after:
Screenshot 2023-04-28 at 17 34 52

Verify attendance check Before:
Screenshot 2023-04-28 at 09 52 52

Screenshot 2023-04-28 at 09 52 26

Verify attendance check After:
Screenshot 2023-04-28 at 09 51 28

Screenshot 2023-04-28 at 10 28 08

Updated room/seat columns:

Screenshot 2023-04-28 at 10 00 55

### CODE REVIEW

**Review by tobias-lippert — DISMISSED** [review]
Tested in testing session on ts1. Worked as expected.

**Comment by sven0311** [comment]
If all students are signed, but one (or more) students are missing some of the other toggles it shows the Red Cross correctly in the overview, but in the "Verify attendance" view it shows that all students are checked successfully. IMO this should show that the toggle is missing

**Review by TheZoker — APPROVED** [review]
Tested within testing session on TS3 and works as expected!

**Review by rriyaldhi — DISMISSED** [review]
Tested on TS3 during the testing session and works as expected.

**Review by FelixTJDietrich — APPROVED** [review]
Testing on ts3 during testing session. Everything looks good and works as expected.

File: src/main/webapp/i18n/de/exam.json [hunk 1]
```diff
@@ -459,14 +464,15 @@
                 "registrationNumberChecked": "Matrikelnummer wurde erfolgreich geprüft",
                 "registrationNumberNotChecked": "Matrikelnummer wurde nicht überprüft",
                 "registeredStudents": "Angemeldete Studierende",
-                "notCheckedStudents": "Nicht überprüfte Studierende",
+                "notCheckedStudents": "Anzahl der nicht überprüften Studenten",
+                "allStudentsCheckedPassed": "Alle Studenten wurden erfolgreich geprüft",
```
Comment (TheZoker) [inline src/main/webapp/i18n/de/exam.json:468]:
```suggestion
                "allStudentsCheckedPassed": "Alle Studenten wurden erfolgreich überprüft",
```

**Review by TheZoker — CHANGES_REQUESTED** [review]

**Review by TheZoker — DISMISSED** [review]

**Review by rriyaldhi — APPROVED** [review]
Reapprove after translation

**Review by danielbode — COMMENTED** [review]
We are required to use gender neutral "Studierende" instead of "Studenten".

**Review by TheZoker — APPROVED** [review]

**Review by rriyaldhi — APPROVED** [review]

**Review by FelixTJDietrich — APPROVED** [review]

**Pull request merged**